discovery plugin LegacyString calls cause panics

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to a specific Terraform provider, please open it in the provider's own repository. The index of providers is at https://github.com/terraform-providers .
-->

### Terraform Version
<!---
Run `terraform -v` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
0.12.18
...
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file. For security, you can also encrypt the files using our GPG public key.
-->

```hcl

# https://releases.hashicorp.com/terraform/
terraform {
  version = "0.12.17"
}

# Any that aren't staged in plugins/ need to be available here: https://releases.hashicorp.com/
providers {
  # Plugins available on releases.hashicorp.com
  aws      = ["~> 2.12"]
  azurerm  = ["~> 1.29"]
  archive  = ["~> 1.2"]
  dome9    = ["~> 1.14"]
}
...
```

### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.
-->
```
 terraform-bundle package -os=linux -arch=amd64 ./terraform_0.12.hcl
Fetching Terraform 0.12.17 core package...
Fetching 3rd party plugins in directory: ./plugins
- Resolving "dome9" provider (~> 1.14)...
- Checking for provider plugin on https://releases.hashicorp.com...
panic: not a legacy Provider

goroutine 1 [running]:
github.com/hashicorp/terraform/addrs.Provider.LegacyString(...)
        /home/tsmith/go/src/github.com/hashicorp/terraform/addrs/provider.go:46
github.com/hashicorp/terraform/plugin/discovery.(*ProviderInstaller).Get(0xc0004bdc08, 0xc0003ed0a0, 0x5, 0x0, 0x0, 0x0, 0x0, 0xc000010058, 0x1, 0x1, ...)
        /home/tsmith/go/src/github.com/hashicorp/terraform/plugin/discovery/get.go:235 +0x1c42
main.(*PackageCommand).Run(0xc0003a4a40, 0xc000032070, 0x3, 0x3, 0x0)
        /home/tsmith/go/src/github.com/hashicorp/terraform/tools/terraform-bundle/package.go:185 +0xef3
github.com/mitchellh/cli.(*CLI).Run(0xc00032f2c0, 0xc00034db90, 0xe6318c, 0x7)
        /home/tsmith/go/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1da
main.main()
        /home/tsmith/go/src/github.com/hashicorp/terraform/tools/terraform-bundle/main.go:75 +0x2e6
```
### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->

### Expected Behavior
<!--
What should have happened?
-->

terraform bundle should build a bundle

### Actual Behavior
<!--
What actually happened?
-->

terraform bundle panicked and crashed.

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->

1.  terraform-bundle package -os=linux -arch=amd64 ./{hcl file above}

### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->
https://travis-ci.org/hashicorp/terraform/builds/622755699 appears to show the commit adding the call failed tests with a similar error.

### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

- #6017

-->

Samme issue here, with `terraform-bundle` version 0.12.18:
```
terraform-bundle package bundle.hcl
Fetching Terraform 0.12.16 core package...
Fetching 3rd party plugins in directory: ./plugins
plugin: ad (0.2.1)
plugin: aixboms (1.1.1)
plugin: burger (0.0.2)
plugin: foreman (0.3.1)
plugin: pfnetman (0.3.0)
plugin: sap (0.2.0)
plugin: uam (0.0.6)
- Resolving "bigip" provider (0.12.3)...
- Checking for provider plugin on https://releases.hashicorp.com...
panic: not a legacy Provider

goroutine 1 [running]:
github.com/hashicorp/terraform/addrs.Provider.LegacyString(...)
        /work/go/goroot/pkg/mod/github.com/hashicorp/terraform@v0.12.18/addrs/provider.go:46
github.com/hashicorp/terraform/plugin/discovery.(*ProviderInstaller).Get(0xc000665c08, 0xc0003f7035, 0x5, 0x0, 0x0, 0x0, 0x0, 0xc000010050, 0x1, 0x1, ...)
        /work/go/goroot/pkg/mod/github.com/hashicorp/terraform@v0.12.18/plugin/discovery/get.go:235 +0x1c42
main.(*PackageCommand).Run(0xc0003aca40, 0xc0000321a0, 0x1, 0x1, 0x0)
        /work/go/goroot/pkg/mod/github.com/hashicorp/terraform@v0.12.18/tools/terraform-bundle/package.go:185 +0xef3
github.com/mitchellh/cli.(*CLI).Run(0xc0003392c0, 0xc000357bf0, 0xe63eec, 0x7)
        /work/go/goroot/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1da
main.main()
        /work/go/goroot/pkg/mod/github.com/hashicorp/terraform@v0.12.18/tools/terraform-bundle/main.go:75 +0x2e6
```

`terraform-bundle` version 0.12.17 seems to be ok

--- a/Makefile
+++ b/Makefile
@@ -39,10 +39,12 @@ testacc: fmtcheck generate
 	TF_ACC=1 go test $(TEST) -v $(TESTARGS) -mod=vendor -timeout 120m
 
 # e2etest runs the end-to-end tests against a generated Terraform binary
+# and a generated terraform-bundle binary.
 # The TF_ACC here allows network access, but does not require any special
-# credentials since the e2etests use local-only providers such as "null".
+# credentials.
 e2etest: generate
 	TF_ACC=1 go test -mod=vendor -v ./command/e2etest
+	TF_ACC=1 go test -mod=vendor -v ./tools/terraform-bundle/e2etest
 
 test-compile: fmtcheck generate
 	@if [ "$(TEST)" = "./..." ]; then \

--- a/tools/terraform-bundle/package.go
+++ b/tools/terraform-bundle/package.go
@@ -182,7 +182,7 @@ func (c *PackageCommand) Run(args []string) int {
 			} else { //attempt to get from the public registry if not found locally
 				c.ui.Output(fmt.Sprintf("- Checking for provider plugin on %s...",
 					releaseHost))
-				_, _, err := installer.Get(addrs.Provider{Type: name}, constraint)
+				_, _, err := installer.Get(addrs.NewLegacyProvider(name), constraint)
 				if err != nil {
 					c.ui.Error(fmt.Sprintf("- Failed to resolve %s provider %s: %s", name, constraint, err))
 					return 1

